api/src/main/java/io/lumeer/api/model/Collection.java

```diff
@@ -46,7 +46,7 @@ default ResourceType getType() {
 
    void updateAttribute(String attributeId, Attribute attribute);
 
-   void deleteAttribute(String attributeId);
+   void deleteAttribute(String attributeName);
```

Comment:
Don't we pass `attributeId` to this method in `CollectionFacade`. The method interface can stay as you designed it, you just need to change the implementation. Find the attribute to be removed first in order to get the name and then remove it together with its children.